### ISSUE TITLE
NameError: name 'L1' is not defined

### ISSUE THREAD

**Issue by songhongxiang** [cdsousa/SymPyBotics#29]
Hi,

I am trying to run the my code step by step in the Python Shell.
When I put `rbt.calc_base_parms()`, I got the following errors.
Please give me any advice to fix this issue.

generating geometric model
generating kinematic model
generating inverse dynamics code
generating gravity term code
generating coriolis term code
generating coriolis matrix code
generating inertia matrix code
generating regressor matrix code
generating friction term code
done
Traceback (most recent call last):
  File ".\symbotics_test.py", line 22, in <module>
    rbt.calc_base_parms()
  File "D:\Program Files\python3.7\lib\site-packages\sympybotics-1.0.dev0-py3.7.egg\sympybotics\robotmodel.py", line 122, in calc_base_parms
  File "D:\Program Files\python3.7\lib\site-packages\sympybotics-1.0.dev0-py3.7.egg\sympybotics\dynamics\dynamics.py", line 55, in calc_base_parms
  File "D:\Program Files\python3.7\lib\site-packages\sympybotics-1.0.dev0-py3.7.egg\sympybotics\dynamics\dyn_parm_dep.py", line 22, in find_dyn_parm_deps
  File "<string>", line 6, in regressor_func
NameError: name 'L1' is not defined

This is my code:
```python
import sympy
import sympybotics
from sympybotics._compatibility_ import exec_

pi = sympy.pi
q = sympybotics.robotdef.q
L1,L2 = sympy.symbols('L1,L2')


twolink = sympybotics.RobotDef('twolink',[(0,0,0,q),
                                         (0,L1,0,q)],
                              dh_convention='modified')

twolink.frictionmodel = {'Coulomb', 'viscous'}
twolink.gravityacc = sympy.Matrix([0.0, 0.0, 'g'])
rbt = sympybotics.RobotDynCode(twolink, verbose=True)

rbt.calc_base_parms()
rbt.dyn.baseparms
```

**Comment by songhongxiang** [comment]
Maybe I have found the answer.
In `robotmodel.py` file:
```py
    def calc_base_parms(self, verbose=False):

        q_subs = {q: 'q[%d]' % i for i, q in enumerate(self.rbtdef.q)}
        q_subs.update(
            {dq: 'dq[%d]' % i for i, dq in enumerate(self.rbtdef.dq)})
        q_subs.update(
            {ddq: 'ddq[%d]' % i for i, ddq in enumerate(self.rbtdef.ddq)})
        func_def_regressor = code_to_func(
            'python', self.H_code, 'regressor', 'regressor_func',
            ['q', 'dq', 'ddq'], q_subs)
```
`func_def_regressor ` is an automatically generated program,but the parameters are missing when executing the program.I tried to increase the required parameters, but I'm failed.

When the link length parameter is a number, the program will not cause problems.

**Comment by cdsousa** [comment]
This is the same problem as https://github.com/cdsousa/SymPyBotics/issues/23

The dynamics part is not supposed to work with symbols, you have to replace those 'L1', 'L2' by the actual numeric values!

**Comment by songhongxiang** [comment]
Thank you for your prompt reply




------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Cristóvão Duarte Sousa"<notifications@github.com&gt;;
发送时间:&nbsp;2020年4月26日(星期天) 晚上6:29
收件人:&nbsp;"cdsousa/SymPyBotics"<SymPyBotics@noreply.github.com&gt;;
抄送:&nbsp;"🐳"<562047964@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [cdsousa/SymPyBotics] NameError: name 'L1' is not defined (#29)





 
Closed #29.

**Comment by ohmyjesus** [comment]
> This is the same problem as #23
> 
> The dynamics part is not supposed to work with symbols, you have to replace those 'L1', 'L2' by the actual numeric values!

ok, thanks very much!
